client/components/domains/registrant-extra-info/fr-form.jsx

```diff
@@ -160,15 +165,16 @@ class RegistrantExtraInfoForm extends React.PureComponent {
 
 	renderPersonalFields() {
 		const translate = this.props.translate;
+		const extra = this.props.contactDetails.extra;
 		const screenReaderText = 'screen-reader-text';
 		const {
 			countryOfBirth,
-			dobDays,
-			dobMonths,
 			dobYears,
+			dobMonths,
+			dobDays,
 			placeOfBirth,
 			postalCodeOfBirth,
-		} = { ...emptyValues, ...this.props.values };
+		} = { ...emptyValues, ...extra, ...this.state };
```

Comment:
Such concatenation might be wasteful if within the `render` stack. `lodash#defaults` could help.